clients/src/main/java/org/apache/kafka/common/Uuid.java

```diff
@@ -151,19 +151,22 @@ private byte[] getBytesFromUuid() {
         return uuidBytes.array();
     }
 
+    /**
+     * Compares this Uuid with another given Uuid.
+     * <p>
+     * This implementation is taken from the compareTo method of the official java.util.UUID implementation.
```

Comment:
We can't take the implementation from OpenJDK, that code has a difference license.

Reply:
@ijuma  We just called it openJDK implementation it's not specific to open JDK since  it's a recent one it's just about a way of comparison mean the compareTo, the same question can be asked when it comes to implement equals and hashCode for example every developer is getting inspired by the openJDK reference implementation 

@divijvaidya  What are your thoughts ? I think we have a nice topic right here on reference implementations and also Licences 

May be we have to add an entry on LICENCES file as we are using other libraries are referencing them 

I agree it's a very good question @ijuma

Reply:
> May be we have to add an entry on LICENCES file as we are using other libraries are referencing them

That's only allowed when the license is compatible - OpenJDK's license is not.

> the same question can be asked when it comes to implement equals and hashCode for example every developer is getting inspired by the openJDK reference implementation

Please check https://github.com/apache/kafka/blob/trunk/CONTRIBUTING.md. It states clearly that you can only contribute your own work.

Reply:
@ijuma  Thanks so much for the review, please give me some time and I will come with my own optimisation, may be I need to stick on the early one I provided, I will update come back soon with some, I think I need some learnings to do this pull request and make it in a better shape